src/lib/logger/impl.ml

```diff
@@ -83,17 +83,33 @@ module Message = struct
   type t =
     { timestamp: Time.t
     ; level: Level.t
-    ; source: Source.t
+    ; source: Source.t option
```

Comment:
I'm not sure if I'm a fan of this solution. Would prefer adding a custom variant to `Source.t` (with a custom yojson implementation) over making source optional.

Reply:
I need to tweak the `Message` yojson impl to elide the field or not, otherwise we end up with a dummy `"source":{}` (or whatever encoding) field in every message. I could make `Source` a variant type anyway, but I don't think there's an advantage here?

Reply:
I would personally prefer `"source": null` to completely eliding `"source"` from the object keys. But I guess it's not much different; it's just a tradeoff between how the data is represented and how the type is represented. No need to change this.